pymarkdown/plugins/rule_md_050.py

```diff
@@ -0,0 +1,88 @@
+"""
+Module to implement a plugin that validates references based on configurable regexes.
+"""
+import re
+from typing import cast
+
+from pymarkdown.inline_markdown_token import ReferenceMarkdownToken
+from pymarkdown.markdown_token import MarkdownToken
+from pymarkdown.plugin_manager.plugin_details import PluginDetails
+from pymarkdown.plugin_manager.plugin_scan_context import PluginScanContext
+from pymarkdown.plugin_manager.rule_plugin import RulePlugin
+
+
+class RuleMd050(RulePlugin):
+    """
+    Class to implement a plugin that validates references based on configurable regexes.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.__regexes = []
+
+    EXTERNAL_LINK_RE = re.compile("^(.+):.*$")
+
+    def get_details(self) -> PluginDetails:
+        """
+        Get the details for the plugin.
+        """
+        return PluginDetails(
+            plugin_name="refs-regex",
+            plugin_id="MD050",
+            plugin_enabled_by_default=False,
+            plugin_description="Reference should match regular expressions",
+            plugin_version="0.0.1",
+            plugin_interface_version=1,
+            plugin_url="https://github.com/jackdewinter/pymarkdown/blob/main/docs/rules/rule_md050.md",
+            plugin_configuration="regexes",
+        )
+
+    def initialize_from_config(self) -> None:
+        """
+        Event to allow the plugin to load configuration information.
+        """
+        self.__regexes = [
```

Comment:
I would prefer the while-loop here with a simple `selfe.__regexes.append(...)` and remove the `[ ... for ... ]` construct.